src/main/java/io/appium/java_client/pagefactory/AppiumElementLocatorFactory.java

```diff
@@ -16,28 +16,51 @@
 
 package io.appium.java_client.pagefactory;
 
+import java.lang.reflect.AnnotatedElement;
 import java.lang.reflect.Field;
 
+import io.appium.java_client.pagefactory.bys.builder.AppiumByBuilder;
+import io.appium.java_client.pagefactory.locator.CacheableElementLocatorFactory;
+import io.appium.java_client.pagefactory.locator.CacheableLocator;
+import org.openqa.selenium.By;
 import org.openqa.selenium.SearchContext;
-import org.openqa.selenium.support.pagefactory.ElementLocator;
-import org.openqa.selenium.support.pagefactory.ElementLocatorFactory;
+import org.openqa.selenium.WebDriver;
 
-class AppiumElementLocatorFactory implements ElementLocatorFactory {
-    private final SearchContext searchContext;
+class AppiumElementLocatorFactory implements CacheableElementLocatorFactory {
+	private final SearchContext searchContext;
 	private final TimeOutDuration timeOutDuration;
+    private final WebDriver originalWebDriver;
+    private final AppiumByBuilder builder;
 
 	public AppiumElementLocatorFactory(SearchContext searchContext,
-			TimeOutDuration timeOutDuration) {
+									   TimeOutDuration timeOutDuration,
+                                       WebDriver originalWebDriver,
+                                       AppiumByBuilder builder) {
 		this.searchContext = searchContext;
+        this.originalWebDriver = originalWebDriver;
 		this.timeOutDuration = timeOutDuration;
+        this.builder = builder;
 	}
-	
-	public AppiumElementLocatorFactory(SearchContext searchContext) {
-		this(searchContext, new TimeOutDuration(AppiumFieldDecorator.DEFAULT_IMPLICITLY_WAIT_TIMEOUT,
-				AppiumFieldDecorator.DEFAULT_TIMEUNIT));
-	}	
-
-	public ElementLocator createLocator(Field field) {
-		return new AppiumElementLocator(searchContext, field, timeOutDuration);
+
+	public CacheableLocator createLocator(Field field) {
+        return this.createLocator((AnnotatedElement) field);
 	}
+
+    @Override
+    public CacheableLocator createLocator(AnnotatedElement annotatedElement) {
+        TimeOutDuration customDuration;
+        if (annotatedElement.isAnnotationPresent(WithTimeout.class)){
+            WithTimeout withTimeout = annotatedElement.getAnnotation(WithTimeout.class);
+            customDuration = new TimeOutDuration(withTimeout.time(), withTimeout.unit());
+        }
+        else
```

Comment:
I think if/else should always use brackets, even for single lines. This is a style suggestion.